burn-core/src/lr_scheduler/step.rs

```diff
@@ -0,0 +1,87 @@
+use burn_tensor::backend::Backend;
+
+use crate as burn;
+
+use super::LrScheduler;
+use crate::{config::Config, LearningRate};
+
+/// Configuration to create a [step](StepLrScheduler) learning rate scheduler.
+#[derive(Config)]
+pub struct StepLrSchedulerConfig {
+    /// The initial learining
+    init_lr: LearningRate,
+    /// Period of learning rate decay
+    #[config(default = 10)]
+    step_size: i64,
+    /// Multiplicative factor of learning rate decay
+    #[config(default = 0.1)]
+    gamma: f64,
+}
+
+/// Decays the learning rate of each parameter group by gamma every step_size epochs.
+#[derive(Clone, Debug)]
+pub struct StepLrScheduler {
+    init_lr: LearningRate,
+    step_size: i64,
+    gamma: f64,
+    step: f64,
```

Comment:
Any reason why the step is `f64` instead of unsigned int like `usize`?

Reply:
I think in the state, it's just to avoid casting.

Reply:
But in `to_record` the step is actually cast as `usize`, and then in `load_record` cast back as `f64`. Maybe I'm missing something, but it just seems like we could keep it `usize` all the way and avoid casting, no?

Reply:
But it's not casted in the `step` function 😅

Reply:
We could use `usize` but I would also update it in the Noam scheduler for consistency.

Reply:
> But it's not casted in the `step` function 😅

Not when incremented, but when it is actually used it seems that it is also casted:

```rust
self.gamma.powf((self.step as i64 / self.step_size) as f64)
```

> We could use `usize` but I would also update it in the Noam scheduler for consistency.

Ahhhhh so that's where it comes from! I didn't realize that was the case.